packages/amplify_flutter/lib/categories/amplify_analytics_category.dart

```diff
@@ -23,8 +23,15 @@ class AnalyticsCategory {
   Future<void> addPlugin(AnalyticsPluginInterface plugin) async {
     //TODO: Allow for multiple plugins to work simultaneously
     if (plugins.length == 0) {
-      plugins.add(plugin);
-      await plugin.addPlugin();
+      try {
+        await plugin.addPlugin();
+        plugins.add(plugin);
+      } on AmplifyAlreadyConfiguredException catch (e) {
+        plugins.add(plugin);
+      } on PlatformException catch (e) {
+        throw AmplifyException.fromMap(
+            Map<String, String>.from(e.details));
+      }
```

Comment:
nit: you can add `plugins.add(plugin)` after the complete try-catch just once.

Reply:
I chose to only re-add the plugin on `AmplifyAlreadyConfiguredException`,  in order ensure a fast failure if some other error gets thrown platform-side with addPlugin.